testing/upgrader.go

```diff
@@ -0,0 +1,69 @@
+// Package testing provides a stub implementation that can be used for
+// simplified testing of applications that normally use tableflip.
+// It is also helpful for allowing projects that use tableflip
+// able to run on Windows, which does not support tableflip.
+package testing
+
+import (
+	"context"
+	"errors"
+)
+
+// Upgrader has all the methods of tableflip.Upgrader, but they don't
+// actually do anything special.
+type Upgrader struct {
+	*Fds
+
+	exitC chan struct{}
+}
+
+// New creates a new stub Upgrader.
+//
+// Unlike the real version, this can be called many times.
+func New() (*Upgrader, error) {
```

Comment:
How do you feel about just using `Upgrader{}` in calling code? Without the channel you don't really need New.

Reply:
The only reason I'd leave it like this is because it looks the same as the actual `Upgrader`. I think the visual symmetry between the methods is better. That way we don't need to confuse people with two different ways of doing it.

Compare:

```
upg, err := tableflip.New(tableflip.Options{
	PIDFile: *pidFile,
})
if errors.Is(err, tableflip.ErrNotSupported) {
	upg, _ = testing.New()
} else if err != nil {
	panic(err)
}
```

vs

```
upg, err := tableflip.New(tableflip.Options{
	PIDFile: *pidFile,
})
if errors.Is(err, tableflip.ErrNotSupported) {
	upg = &testing.Upgrader{}
} else if err != nil {
	panic(err)
}
```

Reply:
Well, you can't give testing.New() any options, and it also never returns an error? I'm not too fussed about this though.

Reply:
I thought of allowing it to take the same `tableflip.Options` argument and throwing an error if it is actually passed in (since the options aren't supported), but I decided not to bother.